### ISSUE TITLE
Mac  fails to build on `0.3.0` but builds perfectly on `master` (96e8a36)

### ISSUE THREAD

**Issue by TG-Techie** [embedded-graphics/simulator#43]
## The Error 
```bash
... ustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/Users/jonahym/Documents/tg-watch-v3/rs-framework/target/debug/deps/rs_framework-4a555f6c32a3b6b0" "-Wl,-dead_strip" "-nodefaultlibs"
  = note: ld: framework not found Security
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
## test case
Building the [themes.rs](https://github.com/embedded-graphics/simulator/blob/96e8a36/examples/themes.rs) example from this repo as a binary crate.

with `Cargo.toml`
```toml
# ...
[dependencies]
embedded-graphics = "0.7.1"
embedded-graphics-simulator = "0.3.0"
```

## Recreation
recreate this with the above toml file, copying `themes.rs` into a fresh `main.rs` and `cargo build` or `cargo run`.

# Debug steps
- Try adding the appropriate (and unappropriate) values to `$LIBRARY_PATH` as recommended. I confirmed this worked by compiling a `c` file with an sdl2 include, ex: `clang sdl_link_test.c -LSDL2`.
- Try rust-sdl2's bundled reature with `println!("cargo:rustc-link-arg=-Wl,-rpath,@loader_path");`
- Try adding `usr/local/lib` and `usr/local/Cellar` to teh build search with `println!(r"cargo:rustc-link-search=...");`
- Try building just sdl2 (which worked...)
- Try cloning and buiulding this repo on its own (which **Also** worked!?)
- Try yanking the source files out of the repo and wuiding them in along side `themes.rs` (I'm Sensing a pattern here...)

# The Solution
Switch the main! While I was debugging I happened to clone `master` (96e8a36)  instead of `0.3.0` and it built! magically!

Here is my final cargo .toml that finally builds and works like a treat!
```toml
# ...
[dependencies]
embedded-graphics = "0.7.1"

[dependencies.embedded-graphics-simulator]
    git = "https://github.com/embedded-graphics/simulator"
    rev = "96e8a36" 
```

# Conclusion / Speculation
With some review, it looks like there's only one really pertient PR since the [`0.3.0` to `96e8a36`](https://github.com/embedded-graphics/simulator/compare/v0.3.0...96e8a36).
PR #39 upgrades from rust 2018 to rust 2021, I suspect that (somehow) interacts with newer rust compiler versions.

# Suggested Actions
Maybe it would make sense to add a check or warning regarding minimum version? I defer to your expertise.

# Misc
## PR #38
Thank you for adding this (#38), it is a great touch and helps sdl/embedded-graphics newcomers like myself get started.

## Enviroment
- macOS Version 12.4 (21F79) on a 2017 Quad-Core Intel Core i7
```bash
rust --version # rustc 1.61.0 (fe5b13d68 2022-05-18)
```
```bash
cargo --version # cargo 1.61.0 (a028ae42f 2022-04-29)
```
- re-linking sdl2 links 10 files from  /usr/local/Cellar/sdl2/2.0.22 Into /usr/local/lib
```bash
brew unlink sdl2 && brew link sdl2
```

### extra misc
(I apologize if this is slightly off-format, I had to write this two times due to a browser bug... yes `<insert safari hate>`)

~ TG-Techie

**Comment by TG-Techie** [comment]
Or releasing a patch?

**Comment by jamwaffles** [comment]
I've just released 0.4.0 with an MSRV of 1.61. Please give that a try and if it doesn't work, feel free to reopen this issue. Thanks for your patience!
